### PR TITLE
Adding company account interceptor plus tests.

### PR DESCRIPTION
**Pull request by sroberts2**
Added the CompanyAccountInterceptor.
Added findById method to CompanyAccount service.
Added the relevant unit tests.

Helps resolve: SFA-509

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            Object handler) {
+
+        final Map<String, Object> debugMap = new HashMap<String, Object>();
+        debugMap.put("request_method", request.getMethod());
+
+        HttpSession session = request.getSession();
+        Transaction transaction = (Transaction) session
+                .getAttribute(AttributeName.TRANSACTION.getValue());
+        if (transaction == null) {
+            LOGGER.debugRequest(request,
```
Comment (macreman) [inline src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java]: errorRequest for internal server error case

File: src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ * The CompanyAccountInterceptor interceptor validates that the Company account is in the correct
+ * state for the calling request. It insures a Company account resource exists against the
+ * transaction, checks the account ID is the same as the one that is referenced in the relevant
+ * Transaction. This interceptor will fail if the Transaction interceptor has not run prior to its
+ * own execution.
+ */
+@Component
+public class CompanyAccountInterceptor extends HandlerInterceptorAdapter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("company-accounts.api.ch.gov.uk");
```
Comment (cburns93) [inline src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java]: Use `APPLICATION_NAME_SPACE` static variable in CompanyAccountsApplication.java?

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java [hunk 3]
```diff
@@ -26,28 +27,27 @@
 @Service
 public class CompanyAccountServiceImpl implements CompanyAccountService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger("company-accounts.api.ch.gov.uk");
```
Comment (cburns93) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java]: same here

File: src/test/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptorTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.servlet.HandlerMapping;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.Kind;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountDataEntity;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
+import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
+import uk.gov.companieshouse.api.accounts.transaction.Resources;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+@ExtendWith(MockitoExtension.class)
```
Comment (cburns93) [inline src/test/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptorTest.java]: no whitespace between any of these lines? best to add it for separation

File: src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java (hunk 1 above)
Comment (sroberts2) [reply src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java]: For a LOGGER.errorRequest the method header is: 
public void errorRequest(HttpServletRequest request, Exception e, Map<String, Object> data).  

I haven't got access to an Exception to pass it what do you suggest? Incidentally I was copying the TransactionAuthenticationInterceptor in its pattern of:

if (authorisedUser == null) { 
LOGGER.debugRequest(request, "TransactionAuthenticationInterceptor error: no authorised identity", null);
response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
return false;
}
 
We should change them in Tandem given we find a solution.

Comment (macreman) [reply src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java]: Good catch in transaction api case. As discussed that should be using errorRequest too and we can pass null in and the library won't blow up so I'd go for that

File: src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            Object handler) {
+
+        final Map<String, Object> debugMap = new HashMap<String, Object>();
+        debugMap.put("request_method", request.getMethod());
+
+        HttpSession session = request.getSession();
+        Transaction transaction = (Transaction) session
+                .getAttribute(AttributeName.TRANSACTION.getValue());
+        if (transaction == null) {
+            LOGGER.errorRequest(request, null, debugMap);
```
Comment (macreman) [inline src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java]: Looking at this in a little more detail, this is fine to do but gives no "message" field in the logs so I think we need to be explicitly setting that in the line above i.e.
`debugMap.put("message", "CompanyAccountInterceptor error: no transaction in request session");`

**Pull request merged**